benchmark/simulate_hgts.py

```diff
@@ -0,0 +1,627 @@
+                             orthologous_groups,
+                             orthologous_rep_prob,
+                             percentage_hgts,
+                             log_f):
+    """Simulate orthologous replacement HGT.
+
+    Notes
+    -----
+    Algorithm: Using list of orthologous genes between donor and recipient
+    genomes
+
+    Parameters
+    ----------
+    genes_donor: dictionary
+    seq_donor: skbio.sequence.Sequence
+    genes_recip: dictionary
+    seq_recip: skbio.sequence.Sequence
+    sequence_ids: dictionary
+    orthologous_groups: list of lists
+    orthologous_rep_prob: float
+    percentage_hgts: float
+    log_f: file descriptor
+
+    Returns
+    -------
+    seq_recip: skbio.sequence.Sequence
+        recipient genome sequence with HGTs
+    """
+    # number of HGTs to simulate
+    num_hgts = int(percentage_hgts*orthologous_rep_prob*len(genes_recip))
+    if num_hgts < 1:
+        num_hgts = 1
+    num_orthogroups = len(orthologous_groups)
+    idx = random.sample(list(range(0, num_orthogroups)), num_hgts)
+    log_f.write("#type\tdonor\tstart\tend\trecipient\tnew label "
+                "recipient\tstart\tend\tstrand\n")
+    for x in range(0, num_hgts):
+        orthogroup = orthologous_groups[idx[x]]
+        substitute_genes = ['*', '*']
+        # randomly select two orthologous genes from the same family
+        # representing the donor and recipient genomes
+        while '*' in substitute_genes:
+            idx2 = random.randrange(0, len(orthogroup))
+            gene = orthogroup[idx2]
+            if (gene.startswith('0') and substitute_genes[0] == '*'):
+                substitute_genes[0] = sequence_ids[gene]
+            elif (gene.startswith('1') and substitute_genes[1] == '*'):
+                substitute_genes[1] = sequence_ids[gene]
+        # match donor and recipient gene labels to results output by
+        # OrthoFinder (in sequence_ids)
+        gene_donor_label = None
+        gene_recip_label = None
+        if substitute_genes[0] in genes_donor:
+            gene_donor_label = substitute_genes[0]
+            gene_recip_label = substitute_genes[1]
+        elif substitute_genes[1] in genes_donor:
+            gene_donor_label = substitute_genes[1]
+            gene_recip_label = substitute_genes[0]
+        else:
+            raise ValueError("Gene %s and %s are not in donor genome" % (
+                substitute_genes[0], substitute_genes[1]))
+        # rename recipient orthologous gene to donor's
+        hgt_gene = "%s_hgt_o" % gene_donor_label
+        genes_recip[hgt_gene] = genes_recip.pop(gene_recip_label)
+        # replace recipient gene (translated sequence) with donor's
+        genes_recip[hgt_gene][0] = genes_donor[gene_donor_label][0]
+        # update end position of HGT gene (as it can be shorter/longer than
+        # the recipient gene replaced)
+        genes_recip[hgt_gene][2] =\
+            genes_recip[hgt_gene][1] + len(genes_recip[hgt_gene][0])*3
```

Comment:
codon.

We need to translate codons into nucleotides in order to insert the gene into the respective genome.

Added a note in the code, thanks!